main.go

```diff
@@ -24,9 +24,15 @@
 package main
 
 import (
+	"os"
+
 	"github.com/coreos/etcd/etcdmain"
+	"github.com/coreos/etcd/migrate/starter"
 )
 
 func main() {
+	if os.Getenv("ETCD_START_DESIRED_VERSION") != "" {
```

Comment:
A better name for this would be "ETCD_ALLOW_LEGACY_MODE". You might want to use strconv.ParseBool() to parse the string.